scripts/bdc/xml_utils.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+"""
+Utility functions for handling NaN values in XML generation process.
+With improved logging and study ID handling.
+"""
+
+import logging
+import os
+import pandas as pd
+import numpy as np
+import sys
+from xml_generator import process_study
+from xml_generator import read_gen3_metadata
+
+def fix_nan_values(df):
+    """Replace NaN values in DataFrame with appropriate empty values"""
+    # Make a copy to avoid modifying the original
+    df_fixed = df.copy()
+    
+    # Replace NaN with empty string for object/string columns
+    for col in df_fixed.select_dtypes(include=['object']).columns:
+        df_fixed[col] = df_fixed[col].fillna('')
+    
+    # Replace NaN with appropriate values for numeric columns 
+    for col in df_fixed.select_dtypes(include=['number']).columns:
+        df_fixed[col] = df_fixed[col].fillna(0)
```

Comment:
This might be a mistake if we're talking about actual values (i.e. a NA is not equal to a 0 when measuring heart rate, say), but I think this will only be used on metadata? Still, I'd be more comfortable with leaving this as NA or replacing it with `999` or something weird to make this clearer. In any case, this behavior should be documented.